lib/application/coordinator/coordinator_router_delegate.dart

```diff
@@ -35,12 +33,9 @@ class CoordinatorRouterDelegate extends RouterDelegate<AppPath>
     );
   }
 
+  // Ignore calls to this method, as `RouteInformationProvider` should be directly injected in the root `MaterialApp`.
   @override
-  Future<void> setInitialRoutePath(AppPath configuration) {
-    // TODO(matuella): This doesn't seems right, but because we are passing our own `RouteInformationProvider` in the
-    // root `MaterialApp`, this will be called once again, thus resetting the navigation.
-    return SynchronousFuture(null);
-  }
+  Future<void> setInitialRoutePath(AppPath configuration) => SynchronousFuture(null);
```

Comment:
No, because this is called by the Flutter framework, although I have to add that I'm not completely sure on this approach, because of how we are using the Nav 2.0 today. Parsing the according `AppPath` and returning it here creates an unexpected side-effect that resets the navigation, because we are already setting it directly through `RouteInformationProvider`. No docs on it though, can't confirm.